chainer/dataset/dataset_mixin.py

```diff
@@ -0,0 +1,46 @@
+class DatasetMixin(object):
+
+    """Default implementation of dataset indexing.
+
+    DatasetMixin provides the :meth:`__getitem__` method using the per-example
+    addressing method :meth:`get_example` given by the implementation. This
+    mixin makes it easy to implement a new dataset that does not support
+    efficient slicing.
+
+    """
+    def __getitem__(self, index):
+        """Returns an example or a sequence of examples.
+
+        It implements the standard Python indexing. It uses the
+        :meth:`get_example` method by default, but it may be overridden by the
+        implementation to improve the slicing performance.
```

Comment:
Add `for example` because the developer can override this method for another reason.